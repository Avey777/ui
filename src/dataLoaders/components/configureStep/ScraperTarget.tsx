// Libraries
import React, {PureComponent, ChangeEvent} from 'react'

// Components
import {
  Form,
  Input,
  Columns,
  Grid,
  ComponentSize,
  InputType,
  ComponentStatus,
} from 'src/clockface'
import BucketDropdown from 'src/dataLoaders/components/BucketsDropdown'
import {Bucket} from 'src/api'

interface Props {
  bucket: string
  buckets: Bucket[]
  onSelectBucket: (bucket: Bucket) => void
  onChangeURL: (value: string) => void
  url: string
}

export class ScraperTarget extends PureComponent<Props> {
  constructor(props: Props) {
    super(props)
  }

  public render() {
    const {onSelectBucket, url, bucket, buckets} = this.props
    return (
      <Grid>
        <Grid.Row>
          <Grid.Column widthXS={Columns.Eight} offsetXS={Columns.Two}>
            <Form.Element label="Bucket">
              <BucketDropdown
                selected={bucket}
                buckets={buckets}
                onSelectBucket={onSelectBucket}
              />
            </Form.Element>
          </Grid.Column>
          <Grid.Column widthXS={Columns.Eight} offsetXS={Columns.Two}>
            <Form.Element
              label="Target URL"
              errorMessage={this.urlEmpty && 'target URL is empty'}
            >
              <Input
                type={InputType.Text}
                value={url}
                onChange={this.handleChangeURL}
                titleText="Target URL"
                size={ComponentSize.Medium}
                autoFocus={true}
                status={this.urlStatus}
              />
            </Form.Element>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    )
  }

  private get urlStatus(): ComponentStatus {
    if (this.urlEmpty) {
      return ComponentStatus.Error
    }
    return ComponentStatus.Default
  }

  private get urlEmpty(): boolean {
    return !this.props.url
  }

  private handleChangeURL = (e: ChangeEvent<HTMLInputElement>) => {
    const value = e.target.value
    this.props.onChangeURL(value)
  }
}

export default ScraperTarget
