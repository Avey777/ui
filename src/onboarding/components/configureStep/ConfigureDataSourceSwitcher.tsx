// Libraries
import React, {PureComponent} from 'react'

// Components
import {ErrorHandling} from 'src/shared/decorators/errors'
import DataStreaming from 'src/onboarding/components/configureStep/streaming/DataStreaming'

// Types
import {DataSource} from 'src/types/v2/dataLoaders'

export interface Props {
  dataSources: DataSource[]
  currentIndex: number
  org: string
  username: string
  bucket: string
}

@ErrorHandling
class ConfigureDataSourceSwitcher extends PureComponent<Props> {
  public render() {
    const {org, bucket, username} = this.props

    switch (this.configurationStep) {
      case 'Listening':
        return <DataStreaming org={org} username={username} bucket={bucket} />
      case 'CSV':
      case 'Line Protocol':
      default:
        return <div>{this.configurationStep}</div>
    }
  }

  private get configurationStep() {
    const {currentIndex, dataSources} = this.props

    if (currentIndex === dataSources.length) {
      return 'Listening'
    }

    return dataSources[currentIndex].name
  }
}

export default ConfigureDataSourceSwitcher
