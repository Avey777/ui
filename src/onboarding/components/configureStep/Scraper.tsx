// Libraries
import React, {PureComponent, ChangeEvent} from 'react'

// Components
import {
  Form,
  Input,
  Columns,
  Grid,
  ComponentSize,
  MultipleInput,
  Dropdown,
  InputType,
} from 'src/clockface'

interface Props {
  bucket: string
  dropdownBuckets: JSX.Element[]
  onChooseInterval: (value: string) => void
  onDropdownHandle: (value: string) => void
  onAddRow: (item: string) => void
  onRemoveRow: (item: string) => void
  onUpdateRow: (index: number, item: string) => void
  tags: Array<{name: string; text: string}>
}

interface State {
  intervalInput: string
}

export class Scraper extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props)

    this.state = {
      intervalInput: '',
    }
  }
  public render() {
    const {intervalInput} = this.state
    const {
      bucket,
      dropdownBuckets,
      onDropdownHandle,
      onAddRow,
      onRemoveRow,
      onUpdateRow,
      tags,
    } = this.props

    return (
      <Grid>
        <Grid.Row>
          <Grid.Column
            widthXS={Columns.Six}
            widthMD={Columns.Five}
            offsetMD={Columns.One}
          >
            <Form.Element label="Interval">
              <Input
                type={InputType.Text}
                value={intervalInput}
                onChange={this.handleChange}
                titleText="Interval"
                size={ComponentSize.Medium}
                autoFocus={true}
              />
            </Form.Element>
          </Grid.Column>
          <Grid.Column
            widthXS={Columns.Six}
            widthMD={Columns.Five}
            offsetMD={Columns.One}
          >
            <Form.Element label="Bucket">
              <Dropdown selectedID={bucket} onChange={onDropdownHandle}>
                {dropdownBuckets}
              </Dropdown>
            </Form.Element>
          </Grid.Column>
          <Grid.Column
            widthXS={Columns.Twelve}
            widthMD={Columns.Ten}
            offsetMD={Columns.One}
          >
            <MultipleInput
              onAddRow={onAddRow}
              onDeleteRow={onRemoveRow}
              onEditRow={onUpdateRow}
              tags={tags}
              title={'Add URL'}
              helpText={''}
            />
          </Grid.Column>
        </Grid.Row>
      </Grid>
    )
  }

  private handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    const value = e.target.value
    this.setState({intervalInput: value})
    this.props.onChooseInterval(value)
  }
}
export default Scraper
