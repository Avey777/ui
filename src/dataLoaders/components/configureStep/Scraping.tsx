// Libraries
import React, {PureComponent} from 'react'
import {connect} from 'react-redux'
import _ from 'lodash'

// Components
import {Form, ComponentStatus} from 'src/clockface'
import FancyScrollbar from 'src/shared/components/fancy_scrollbar/FancyScrollbar'
import OnboardingButtons from 'src/onboarding/components/OnboardingButtons'
import ScraperTarget from 'src/dataLoaders/components/configureStep/ScraperTarget'

// Actions
import {
  setScraperTargetBucket,
  setScraperTargetURL,
  saveScraperTarget,
} from 'src/dataLoaders/actions/dataLoaders'
import {setBucketInfo} from 'src/dataLoaders/actions/steps'

// Types
import {Bucket} from '@influxdata/influx'
import {AppState} from 'src/types/v2/index'

interface OwnProps {
  onClickNext: () => void
  buckets: Bucket[]
}

interface DispatchProps {
  onSetScraperTargetBucket: typeof setScraperTargetBucket
  onSetScraperTargetURL: typeof setScraperTargetURL
  onSaveScraperTarget: typeof saveScraperTarget
  onSetBucketInfo: typeof setBucketInfo
}

interface StateProps {
  scraperBucket: string
  url: string
  currentBucket: string
}

type Props = OwnProps & DispatchProps & StateProps

export class Scraping extends PureComponent<Props> {
  public componentDidMount() {
    const {
      buckets,
      scraperBucket,
      currentBucket,
      onSetScraperTargetBucket,
    } = this.props

    if (!scraperBucket) {
      onSetScraperTargetBucket(currentBucket || _.get(buckets, '0.name', ''))
    }
  }

  public render() {
    const {scraperBucket, onSetScraperTargetURL, url, buckets} = this.props

    return (
      <Form onSubmit={this.handleSubmit}>
        <div className="wizard-step--scroll-area">
          <FancyScrollbar autoHide={false}>
            <div className="wizard-step--scroll-content">
              <h3 className="wizard-step--title">Add Scraper Target</h3>
              <h5 className="wizard-step--sub-title">
                Scrapers collect data from multiple targets at regular intervals
                and to write to a bucket
              </h5>
              <ScraperTarget
                bucket={scraperBucket}
                buckets={buckets}
                onSelectBucket={this.handleSelectBucket}
                onChangeURL={onSetScraperTargetURL}
                url={url}
              />
            </div>
          </FancyScrollbar>
        </div>
        <OnboardingButtons
          autoFocusNext={false}
          nextButtonStatus={this.nextButtonStatus}
          nextButtonText={'Finish'}
        />
      </Form>
    )
  }

  private handleSelectBucket = (bucket: Bucket) => {
    const {onSetBucketInfo, onSetScraperTargetBucket} = this.props
    const {organization, organizationID, id, name} = bucket

    onSetBucketInfo(organization, organizationID, name, id)
    onSetScraperTargetBucket(bucket.name)
  }

  private get nextButtonStatus(): ComponentStatus {
    if (
      this.props.url === '' ||
      !this.props.buckets ||
      !this.props.buckets.length
    ) {
      return ComponentStatus.Disabled
    }
    return ComponentStatus.Default
  }

  private handleSubmit = async () => {
    const {onSaveScraperTarget, onClickNext} = this.props
    await onSaveScraperTarget()
    onClickNext()
  }
}

const mstp = ({
  dataLoading: {
    dataLoaders: {scraperTarget},
    steps: {bucket},
  },
}: AppState): StateProps => {
  return {
    currentBucket: bucket,
    scraperBucket: scraperTarget.bucket,
    url: scraperTarget.url,
  }
}

const mdtp: DispatchProps = {
  onSetScraperTargetBucket: setScraperTargetBucket,
  onSetScraperTargetURL: setScraperTargetURL,
  onSaveScraperTarget: saveScraperTarget,
  onSetBucketInfo: setBucketInfo,
}

export default connect<StateProps, DispatchProps, OwnProps>(
  mstp,
  mdtp
)(Scraping)
