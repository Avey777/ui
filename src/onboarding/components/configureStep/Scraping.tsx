// Libraries
import React, {PureComponent} from 'react'
import {connect} from 'react-redux'

// Components
import {Form, Dropdown} from 'src/clockface'
import FancyScrollbar from 'src/shared/components/fancy_scrollbar/FancyScrollbar'
import OnboardingButtons from 'src/onboarding/components/OnboardingButtons'

// Actions
import {
  setScrapingInterval,
  addScrapingURL,
  removeScrapingURL,
  updateScrapingURL,
  setScrapingBucket,
} from 'src/onboarding/actions/dataLoaders'
import {AppState} from 'src/types/v2/index'
import {SetupParams} from 'src/onboarding/apis'
import Scraper from 'src/onboarding/components/configureStep/Scraper'

interface OwnProps {
  onClickNext: () => void
  onClickBack: () => void
  onClickSkip: () => void
}

interface DispatchProps {
  setScrapingInterval: typeof setScrapingInterval
  setScrapingBucket: typeof setScrapingBucket
  addScrapingURL: typeof addScrapingURL
  removeScrapingURL: typeof removeScrapingURL
  updateScrapingURL: typeof updateScrapingURL
}

interface StateProps {
  interval: string
  bucket: string
  urls: string[]
  setupParams: SetupParams
}

type Props = OwnProps & DispatchProps & StateProps

export class Scraping extends PureComponent<Props> {
  public render() {
    const {bucket, onClickBack, onClickSkip} = this.props
    return (
      <Form onSubmit={this.handleSubmit}>
        <div className="wizard-step--scroll-area">
          <FancyScrollbar autoHide={false}>
            <div className="wizard-step--scroll-content">
              <h3 className="wizard-step--title">Add Scraper</h3>
              <h5 className="wizard-step--sub-title">
                Scrapers collect data from multiple targets at regular intervals
                and to write to a bucket
              </h5>
              <Scraper
                bucket={bucket}
                dropdownBuckets={this.dropdownBuckets}
                onChooseInterval={this.handleChange}
                onDropdownHandle={this.handleBucket}
                onAddRow={this.handleAddRow}
                onRemoveRow={this.handleRemoveRow}
                onUpdateRow={this.handleEditRow}
                tags={this.tags}
              />
            </div>
          </FancyScrollbar>
        </div>
        <OnboardingButtons
          onClickBack={onClickBack}
          onClickSkip={onClickSkip}
          showSkip={true}
          autoFocusNext={true}
          skipButtonText={'Skip'}
        />
      </Form>
    )
  }
  private get dropdownBuckets(): JSX.Element[] {
    const {setupParams} = this.props

    const buckets = [setupParams.bucket]

    // This is a hacky fix
    return buckets.map(b => (
      <Dropdown.Item key={b} value={b} id={b}>
        {b}
      </Dropdown.Item>
    ))
  }

  private handleChange = (value: string) => {
    this.props.setScrapingInterval(value)
  }

  private handleAddRow = (item: string) => {
    this.props.addScrapingURL(item)
  }

  private handleRemoveRow = (item: string) => {
    this.props.removeScrapingURL(item)
  }

  private handleEditRow = (index: number, item: string) => {
    this.props.updateScrapingURL(index, item)
  }

  private get tags(): Array<{name: string; text: string}> {
    const {urls} = this.props
    return urls.map(v => {
      return {text: v, name: v}
    })
  }

  private handleBucket = (bucket: string) => {
    this.props.setScrapingBucket(bucket)
  }

  private handleSubmit = () => {
    this.props.onClickNext()
  }
}

const mstp = ({
  onboarding: {
    steps: {setupParams},
    dataLoaders: {
      scraper: {interval, bucket, urls},
    },
  },
}: AppState) => {
  return {setupParams, interval, bucket, urls}
}

const mdtp: DispatchProps = {
  setScrapingInterval,
  setScrapingBucket,
  addScrapingURL,
  removeScrapingURL,
  updateScrapingURL,
}
export default connect<StateProps, DispatchProps, OwnProps>(
  mstp,
  mdtp
)(Scraping)
