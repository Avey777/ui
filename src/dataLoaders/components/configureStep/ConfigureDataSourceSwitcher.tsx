// Libraries
import React, {PureComponent} from 'react'
import _ from 'lodash'

// Components
import {ErrorHandling} from 'src/shared/decorators/errors'
import LineProtocol from 'src/dataLoaders/components/configureStep/lineProtocol/LineProtocol'
import PluginConfigSwitcher from 'src/dataLoaders/components/configureStep/streaming/PluginConfigSwitcher'
import EmptyDataSourceState from 'src/dataLoaders/components/configureStep/EmptyDataSourceState'
import Scraping from 'src/dataLoaders/components/configureStep/Scraping'

// Actions
import {
  updateTelegrafPluginConfig,
  addConfigValue,
  removeConfigValue,
  setConfigArrayValue,
} from 'src/dataLoaders/actions/dataLoaders'

// Types
import {TelegrafPlugin, DataLoaderType, Substep} from 'src/types/v2/dataLoaders'
import {Bucket} from 'src/api'

export interface Props {
  telegrafPlugins: TelegrafPlugin[]
  substepIndex: Substep
  onUpdateTelegrafPluginConfig: typeof updateTelegrafPluginConfig
  onAddConfigValue: typeof addConfigValue
  onRemoveConfigValue: typeof removeConfigValue
  dataLoaderType: DataLoaderType
  buckets: Bucket[]
  bucket: string
  org: string
  username: string
  onSetConfigArrayValue: typeof setConfigArrayValue
  onClickNext: () => void
  onClickPrevious: () => void
}

@ErrorHandling
class ConfigureDataSourceSwitcher extends PureComponent<Props> {
  public render() {
    const {
      bucket,
      org,
      telegrafPlugins,
      substepIndex,
      dataLoaderType,
      onUpdateTelegrafPluginConfig,
      onAddConfigValue,
      onRemoveConfigValue,
      onSetConfigArrayValue,
      onClickNext,
      onClickPrevious,
      buckets,
    } = this.props

    switch (dataLoaderType) {
      case DataLoaderType.Streaming:
        return (
          <div className="onboarding-step wizard--skippable">
            <PluginConfigSwitcher
              onUpdateTelegrafPluginConfig={onUpdateTelegrafPluginConfig}
              onRemoveConfigValue={onRemoveConfigValue}
              telegrafPlugins={telegrafPlugins}
              substepIndex={substepIndex}
              onAddConfigValue={onAddConfigValue}
              onSetConfigArrayValue={onSetConfigArrayValue}
              onClickNext={onClickNext}
              onClickPrevious={onClickPrevious}
            />
          </div>
        )
      case DataLoaderType.LineProtocol:
        return (
          <div className="onboarding-step">
            <LineProtocol bucket={bucket} org={org} onClickNext={onClickNext} />
          </div>
        )
      case DataLoaderType.Scraping:
        return (
          <div className="onboarding-step">
            <Scraping onClickNext={onClickNext} buckets={buckets} />
          </div>
        )
      case DataLoaderType.CSV:
        return <div>{DataLoaderType.CSV}</div>
      default:
        return <EmptyDataSourceState />
    }
  }
}

export default ConfigureDataSourceSwitcher
