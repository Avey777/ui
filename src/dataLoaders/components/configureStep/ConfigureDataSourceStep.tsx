// Libraries
import React, {PureComponent} from 'react'
import _ from 'lodash'

// Components
import {ErrorHandling} from 'src/shared/decorators/errors'
import ConfigureDataSourceSwitcher from 'src/dataLoaders/components/configureStep/ConfigureDataSourceSwitcher'

// Actions
import {setActiveTelegrafPlugin} from 'src/dataLoaders/actions/dataLoaders'
import {
  updateTelegrafPluginConfig,
  setPluginConfiguration,
  addConfigValue,
  removeConfigValue,
  setConfigArrayValue,
} from 'src/dataLoaders/actions/dataLoaders'

// Types
import {DataLoaderStepProps} from 'src/dataLoaders/components/DataLoadersWizard'
import {TelegrafPlugin, DataLoaderType} from 'src/types/v2/dataLoaders'
import {Bucket} from 'src/api'

export interface OwnProps extends DataLoaderStepProps {
  telegrafPlugins: TelegrafPlugin[]
  onSetActiveTelegrafPlugin: typeof setActiveTelegrafPlugin
  onUpdateTelegrafPluginConfig: typeof updateTelegrafPluginConfig
  onSetPluginConfiguration: typeof setPluginConfiguration
  type: DataLoaderType
  onAddConfigValue: typeof addConfigValue
  onRemoveConfigValue: typeof removeConfigValue
  onSetConfigArrayValue: typeof setConfigArrayValue
  bucket: string
  org: string
  username: string
  buckets: Bucket[]
}

type Props = OwnProps

@ErrorHandling
export class ConfigureDataSourceStep extends PureComponent<Props> {
  constructor(props: Props) {
    super(props)
  }

  public render() {
    const {
      telegrafPlugins,
      type,
      substep,
      onUpdateTelegrafPluginConfig,
      onAddConfigValue,
      onRemoveConfigValue,
      onSetConfigArrayValue,
      bucket,
      org,
      username,
      buckets,
    } = this.props

    return (
      <ConfigureDataSourceSwitcher
        buckets={buckets}
        bucket={bucket}
        org={org}
        username={username}
        telegrafPlugins={telegrafPlugins}
        onUpdateTelegrafPluginConfig={onUpdateTelegrafPluginConfig}
        onAddConfigValue={onAddConfigValue}
        onRemoveConfigValue={onRemoveConfigValue}
        dataLoaderType={type}
        substepIndex={substep}
        onSetConfigArrayValue={onSetConfigArrayValue}
        onClickNext={this.handleNext}
        onClickPrevious={this.handlePrevious}
      />
    )
  }

  private handleNext = async () => {
    const {
      onIncrementCurrentStepIndex,
      onSetActiveTelegrafPlugin,
      onSetPluginConfiguration,
      telegrafPlugins,
      substep,
      currentStepIndex,
      type,
      onExit,
      onSetSubstepIndex,
    } = this.props

    if (type === DataLoaderType.Scraping) {
      onExit()
      return
    }

    if (type === DataLoaderType.Streaming && !_.isNaN(Number(substep))) {
      const index = +substep
      const telegrafPlugin = _.get(telegrafPlugins, `${index}.name`)
      onSetPluginConfiguration(telegrafPlugin)
      onSetActiveTelegrafPlugin('')
      onSetSubstepIndex(currentStepIndex, 'config')
      return
    }

    onIncrementCurrentStepIndex()
  }

  private handlePrevious = () => {
    const {
      type,
      substep,
      currentStepIndex,
      onSetSubstepIndex,
      onDecrementCurrentStepIndex,
    } = this.props

    if (type === DataLoaderType.Streaming) {
      if (substep === 'config') {
        onSetSubstepIndex(+currentStepIndex - 1, 'streaming')
      }
      return
    }

    onDecrementCurrentStepIndex()
  }
}

export default ConfigureDataSourceStep
