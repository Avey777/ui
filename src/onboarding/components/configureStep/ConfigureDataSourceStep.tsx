// Libraries
import React, {PureComponent} from 'react'
import _ from 'lodash'

// Components
import {ErrorHandling} from 'src/shared/decorators/errors'
import {
  Button,
  ComponentColor,
  ComponentSize,
  ComponentStatus,
} from 'src/clockface'
import ConfigureDataSourceSwitcher from 'src/onboarding/components/configureStep/ConfigureDataSourceSwitcher'

// Types
import {OnboardingStepProps} from 'src/onboarding/containers/OnboardingWizard'
import {DataSource, DataLoaderType} from 'src/types/v2/dataLoaders'

export interface Props extends OnboardingStepProps {
  dataSources: DataSource[]
  type: DataLoaderType
}

interface State {
  currentDataSourceIndex: number
}

@ErrorHandling
class ConfigureDataSourceStep extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props)

    this.state = {
      currentDataSourceIndex: 0,
    }
  }

  public render() {
    const {setupParams, dataSources} = this.props

    return (
      <div className="onboarding-step">
        <ConfigureDataSourceSwitcher
          dataSources={dataSources}
          currentIndex={this.state.currentDataSourceIndex}
          org={_.get(setupParams, 'org', '')}
          username={_.get(setupParams, 'username', '')}
          bucket={_.get(setupParams, 'bucket', '')}
        />
        <div className="wizard-button-bar">
          <Button
            color={ComponentColor.Default}
            text="Back"
            size={ComponentSize.Medium}
            onClick={this.handlePrevious}
          />
          <Button
            color={ComponentColor.Primary}
            text="Next"
            size={ComponentSize.Medium}
            onClick={this.handleNext}
            status={ComponentStatus.Default}
            titleText={'Next'}
          />
        </div>
      </div>
    )
  }

  private handleNext = () => {
    const {onIncrementCurrentStepIndex, dataSources} = this.props
    const {currentDataSourceIndex} = this.state

    if (currentDataSourceIndex >= dataSources.length) {
      onIncrementCurrentStepIndex()
    } else {
      this.setState({currentDataSourceIndex: currentDataSourceIndex + 1})
    }
  }

  private handlePrevious = () => {
    const {onDecrementCurrentStepIndex} = this.props
    const {currentDataSourceIndex} = this.state

    if (currentDataSourceIndex === 0) {
      onDecrementCurrentStepIndex()
    } else {
      this.setState({currentDataSourceIndex: currentDataSourceIndex - 1})
    }
  }
}

export default ConfigureDataSourceStep
