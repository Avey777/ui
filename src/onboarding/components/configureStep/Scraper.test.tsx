// Libraries
import React from 'react'
import {shallow} from 'enzyme'

// Components
import {Scraper} from 'src/onboarding/components/configureStep/Scraper'

const setup = (override = {}) => {
  const props = {
    bucket: '',
    dropdownBuckets: [<></>],
    onChooseInterval: jest.fn(),
    onDropdownHandle: jest.fn(),
    onAddRow: jest.fn(),
    onRemoveRow: jest.fn(),
    onUpdateRow: jest.fn(),
    tags: [{name: '', text: ''}],
    ...override,
  }
  const wrapper = shallow(<Scraper {...props} />)

  return {wrapper}
}

describe('Scraping', () => {
  describe('rendering', () => {
    it('renders!', () => {
      const {wrapper} = setup()
      expect(wrapper.exists()).toBe(true)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
