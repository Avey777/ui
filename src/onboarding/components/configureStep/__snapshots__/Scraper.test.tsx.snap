// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scraping rendering renders! 1`] = `
<Grid>
  <GridRow>
    <GridColumn
      offsetMD={1}
      widthMD={5}
      widthXS={6}
    >
      <FormElement
        label="Interval"
      >
        <Input
          autoFocus={true}
          autocomplete="off"
          disabledTitleText="This input is disabled"
          name=""
          onChange={[Function]}
          placeholder=""
          size="md"
          spellCheck={false}
          status="default"
          titleText="Interval"
          type="text"
          value=""
        />
      </FormElement>
    </GridColumn>
    <GridColumn
      offsetMD={1}
      widthMD={5}
      widthXS={6}
    >
      <FormElement
        label="Bucket"
      >
        <Dropdown
          buttonColor="default"
          buttonSize="sm"
          maxMenuHeight={250}
          menuColor="sapphire"
          mode="radio"
          onChange={[MockFunction]}
          selectedID=""
          status="default"
          titleText=""
          wrapText={false}
        />
      </FormElement>
    </GridColumn>
    <GridColumn
      offsetMD={1}
      widthMD={10}
      widthXS={12}
    >
      <MultipleInput
        helpText=""
        onAddRow={[MockFunction]}
        onDeleteRow={[MockFunction]}
        onEditRow={[MockFunction]}
        tags={
          Array [
            Object {
              "name": "",
              "text": "",
            },
          ]
        }
        title="Add URL"
      />
    </GridColumn>
  </GridRow>
</Grid>
`;
